File: lib/layer2_core/dark/e_on_primary_fixed_variant_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EOnPrimaryFixedVariantDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd23;
      case 'standard': return cstd23;
      case 'blue': return cbd23;
      case 'green': return cgd23;
      case 'red': return crd23;
      case 'nothing': return cnod23;
      default: return cstd23;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EOnPrimaryFixedVariantDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnPrimaryFixedVariantDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd23
            AppTheme.STANDARD -> R.color.cstd23
            AppTheme.BLUE -> R.color.cbd23
            AppTheme.GREEN -> R.color.cgd23
            AppTheme.RED -> R.color.crd23
            AppTheme.NOTHING -> R.color.cnod23
        }
    }
}
